### ISSUE TITLE
Build error in rawspeed VC5Decompressor

### ISSUE THREAD

**Issue by esq4** [darktable-org/darktable#9988]
Hello.

I'm building project with _bash ../build.sh --prefix /opt/darktable-dev --build-type Debug_.
As a result:

```
[  6%] Building CXX object lib/darktable/rawspeed/CMakeFiles/rawspeed.dir/src/librawspeed/decompressors/VC5Decompressor.cpp.o
In file included from /usr/include/c++/8/cassert:44,
                 from /media/data1/darktable/src/external/rawspeed/src/librawspeed/decompressors/VC5Decompressor.cpp:39:
/media/data1/darktable/src/external/rawspeed/src/librawspeed/decompressors/VC5Decompressor.cpp: In member function ‘void rawspeed::VC5Decompressor::Wavelet::ReconstructableBand::createLowpassReconstructionTask(bool&)’:
/media/data1/darktable/src/external/rawspeed/src/librawspeed/decompressors/VC5Decompressor.cpp:298:7: error: ‘__PRETTY_FUNCTION__’ not specified in enclosing ‘task’
       assert(highlow.has_value() && lowlow.has_value() &&
       ^~~~~~
/media/data1/darktable/src/external/rawspeed/src/librawspeed/decompressors/VC5Decompressor.cpp:289:9: error: enclosing ‘task’
 #pragma omp task default(none)                                                 \
         ^~~
In file included from /usr/include/c++/8/cassert:44,
                 from /media/data1/darktable/src/external/rawspeed/src/librawspeed/decompressors/VC5Decompressor.cpp:39:
/media/data1/darktable/src/external/rawspeed/src/librawspeed/decompressors/VC5Decompressor.cpp: In member function ‘void rawspeed::VC5Decompressor::Wavelet::ReconstructableBand::createHighpassReconstructionTask(bool&)’:
/media/data1/darktable/src/external/rawspeed/src/librawspeed/decompressors/VC5Decompressor.cpp:324:7: error: ‘__PRETTY_FUNCTION__’ not specified in enclosing ‘task’
       assert(highhigh.has_value() && lowhigh.has_value() &&
       ^~~~~~
/media/data1/darktable/src/external/rawspeed/src/librawspeed/decompressors/VC5Decompressor.cpp:315:9: error: enclosing ‘task’
 #pragma omp task default(none)                                                 \
         ^~~
In file included from /usr/include/c++/8/cassert:44,
                 from /media/data1/darktable/src/external/rawspeed/src/librawspeed/decompressors/VC5Decompressor.cpp:39:
/media/data1/darktable/src/external/rawspeed/src/librawspeed/decompressors/VC5Decompressor.cpp: In member function ‘void rawspeed::VC5Decompressor::Wavelet::ReconstructableBand::createLowHighPassCombiningTask(bool&)’:
/media/data1/darktable/src/external/rawspeed/src/librawspeed/decompressors/VC5Decompressor.cpp:354:7: error: ‘__PRETTY_FUNCTION__’ not specified in enclosing ‘task’
       assert(lowpass.has_value() && highpass.has_value() &&
       ^~~~~~
/media/data1/darktable/src/external/rawspeed/src/librawspeed/decompressors/VC5Decompressor.cpp:345:9: error: enclosing ‘task’
 #pragma omp task default(none)                                                 \
         ^~~
In file included from /usr/include/c++/8/cassert:44,
                 from /media/data1/darktable/src/external/rawspeed/src/librawspeed/decompressors/VC5Decompressor.cpp:39:
/media/data1/darktable/src/external/rawspeed/src/librawspeed/decompressors/VC5Decompressor.cpp: In member function ‘virtual void rawspeed::VC5Decompressor::Wavelet::AbstractDecodeableBand::createDecodingTasks(rawspeed::ErrorLog&, bool&)’:
/media/data1/darktable/src/external/rawspeed/src/librawspeed/decompressors/VC5Decompressor.cpp:585:9: error: ‘__PRETTY_FUNCTION__’ not specified in enclosing ‘task’
         assert(!decodedData.has_value() && "Decoded this band already?");
         ^~~~~~
/media/data1/darktable/src/external/rawspeed/src/librawspeed/decompressors/VC5Decompressor.cpp:577:9: error: enclosing ‘task’
 #pragma omp task default(none) shared(decodedData, errLog, exceptionThrown)    \
         ^~~
make[2]: *** [lib/darktable/rawspeed/CMakeFiles/rawspeed.dir/build.make:687: lib/darktable/rawspeed/CMakeFiles/rawspeed.dir/src/librawspeed/decompressors/VC5Decompressor.cpp.o] Ошибка 1
make[1]: *** [CMakeFiles/Makefile2:2679: lib/darktable/rawspeed/CMakeFiles/rawspeed.dir/all] Ошибка 2
make: *** [Makefile:152: all] Ошибка 2
```
With option _--build-type Release_, the build succeeds.

Darktable Version: master
OS: Debian 10

**Comment by Nilvus** [comment]
Please precise which exact commit you use and be sure to test with last master. A commit had been posted by @LebedevRI about some build issue. See here: https://github.com/darktable-org/darktable/commit/b174c61bef3c50b3bcebf0e3ff0ef3ffa4832202

**Comment by esq4** [comment]
> Please precise which exact commit you use and be sure to test with last master. A commit had been posted by @LebedevRI about some build issue. See here: [b174c61](https://github.com/darktable-org/darktable/commit/b174c61bef3c50b3bcebf0e3ff0ef3ffa4832202)

Yes, the last master.

**Comment by johnny-bit** [comment]
Debug builds are generally not recommended unless you reallyreallyreally know and understand that those may lead to untested/unworkable codepaths since some of them are never tested unless super extra necessary...

That being said, we generally don't like builds failing. However in this case this is not directly darktable issue, but rawspeed one. Please try building rawspeed alone with debug build and if it fails - open new issue here: https://github.com/darktable-org/rawspeed/issues/new/choose

Meanwhile please close this issue.

**Comment by LebedevRI** [comment]
Hm. Well, that's not great. That is yet another reason why [support for pre-gcc-10 needs to be dropped](https://github.com/darktable-org/rawspeed/blob/b1909b8694457bbf5ba94b7c00eb4ad47d3e9e6b/cmake/compiler-versions.cmake#L72-L79),
because as it miraculously happens, that will resolve this issue too:
https://godbolt.org/z/W56YqMxEY

TLDR: if you really need debug build, you need gcc10.

**Comment by johnny-bit** [comment]
Sooo... "debug builds work in 10+" and this issue is non-issue?

**Comment by LebedevRI** [comment]
Yep.

**Comment by ralfbrown** [comment]
GCC10 is only 16 months old.  Dropping support for anything prior to that also means abandoning anyone on an LTS distro from prior to 2021.

**Comment by LebedevRI** [comment]
Both debian stable and latest ubuntu lts come with gcc10

**Comment by johnny-bit** [comment]
Well, it's not really dropping support isn't it? I mean - the CI builds normally for GCC7 now, normal release and relwithdebinfo build quite well with gcc 7+, the recommendation is for 10+ and since debug builds are for debugging anyway (and are sure as hell not recommended unless you really really really need them) then i guess requiring newer GCC is "ok"?

Personally I'd like to see a general recommendation about packages & LTS support. Currently I think we have unofficial "current and previous ubuntu LTS" for "normal" Linux builds. Would be great to make such thing official and do similar policies around msys2/windows and xcode/mac so we know what to expect for compilation.
Another would be an official policy for 'Debug' builds - I remember debug being very not recommended was because houz told me so (at least twice) and then i think there's some comment in code that some debug path is untested and will most likely fail? Anyway - having an official policy for all builds would be great for reliability.

**Comment by LebedevRI** [comment]
> Well, it's not really dropping support isn't it?

Requiring new gcc for debug builds isn't dropping support indeed,
but the plan *is* to drop support completely.

**Comment by johnny-bit** [comment]
> but the plan is to drop support completely.

I'm OK with that as long as the plan is agreed upon and made official :)

**Comment by LebedevRI** [comment]
For rawspeed, i have already agreed upon that and made that official in https://github.com/darktable-org/rawspeed/blob/b1909b8694457bbf5ba94b7c00eb4ad47d3e9e6b/cmake/compiler-versions.cmake#L72-L79

**Comment by johnny-bit** [comment]
Sure, But we ought to put that not only in cmake files, but in readmes too.
